src/events/crawler/scrapers/ESP/index.js

```diff
@@ -1,38 +1,149 @@
+import { LocalDate } from '@js-joda/core';
 import * as fetch from '../../lib/fetch.js';
-import * as parse from '../../lib/parse.js';
 import * as transform from '../../lib/transform.js';
-import * as rules from '../../lib/rules.js';
-
-// Set county to this if you only have state data, but this isn't the entire state
-// const UNASSIGNED = '(unassigned)';
+import * as parse from '../../lib/parse.js';
 
 const scraper = {
   country: 'ESP',
-  url: 'https://opendata.arcgis.com/datasets/48fac2d7de0f43f9af938852e3748845_0.csv',
+  url: 'https://github.com/datadista/datasets/tree/master/COVID%2019',
   priority: 1,
-  // aggregate: 'state', // doesn't seem to be aggregating properly
+  timeseries: true,
+  aggregate: 'state',
+  sources: [
+    {
+      name: 'cases',
+      url: 'https://raw.githubusercontent.com/datadista/datasets/master/COVID%2019/ccaa_covid19_casos.csv',
+      description: 'Cases by date and community'
+    },
+    {
+      name: 'recovered',
+      url: 'https://raw.githubusercontent.com/datadista/datasets/master/COVID%2019/ccaa_covid19_altas.csv',
+      description: 'Recoveries by date and community'
+    },
+    {
+      name: 'deaths',
+      url: 'https://raw.githubusercontent.com/datadista/datasets/master/COVID%2019/ccaa_covid19_fallecidos.csv',
+      description: 'Deaths by date and community'
+    }
+  ],
+  curators: [
+    {
+      name: 'Antonio Delgado', //
+      url: 'https://datadista.com/',
+      twitter: '@adelgado',
+      github: 'adelgadob'
+    }
+  ],
+  maintainers: [
+    {
+      name: 'Herb Caudill', //
+      url: 'https://devresults.com',
+      twitter: '@herbcaudill',
+      github: 'herbcaudill'
+    }
+  ],
   async scraper() {
-    const data = await fetch.csv(this.url);
-    const states = [];
-    for (const row of data) {
-      const state = row.Texto !== undefined ? parse.string(row.Texto) : '';
-      const cases = row.TotalConfirmados !== undefined ? parse.number(row.TotalConfirmados) : 0;
-      const deaths = row.TotalFallecidos !== undefined ? parse.number(row.TotalFallecidos) : 0;
-      const recovered = row.TotalRecuperados !== undefined ? parse.number(row.TotalRecuperados) : 0;
-      if (state !== '') {
-        const data = {
-          state,
-          cases,
-          deaths,
-          recovered
-        };
-        if (rules.isAcceptable(data, null, this._reject)) {
-          states.push(data);
-        }
-      }
+    const isDate = s => s.includes('/');
+
+    const parseDate = s => {
+      // Source data currently uses ddmmyyyy format for dates,
+      // but is (maybe?) planning to switch to yyyymmdd.
+      // See https://github.com/datadista/datasets/issues/14
+      // When/if this happens, replace contents of this function with:
+      // return LocalDate.parse(s)
```

Comment:
I see this issue was closed, and the dates in the source have changed to YYYY-MM-DD

Reply:
👍 done